### PR TITLE
Fix carousel selected state and dots

### PR DESCRIPTION
**Pull request by jasonadkison**
- fixes an issue where the `selected` state was not properly calculated when looping through the first and last images in the set
- fixes the selected carousel dots styling (css specificity issue)
- fixes an issue where the first thumbnail could never have the `selectedTab` classname applied

### CODE REVIEW

**Comment by jasonadkison** [comment]
@dijs please give this a review asap, thanks!

File: src/carousel/Carousel.js [hunk 1]
```diff
@@ -108,6 +108,16 @@ const Carousel = React.forwardRef((props, ref) => {
     return <Fragment key={key}>{slide}</Fragment>
   }
 
+  const onChangeIndex = useCallback((index) => {
+    if (!infinite) {
+      setSelected(index)
+      return
+    }
+
+    const nextSelected = Math.abs(index - count * Math.floor(index / count))
```
Comment (dijs) [inline src/carousel/Carousel.js:117]: Can you please add some comments explaining this math.

**Review by dijs — CHANGES_REQUESTED** [review]
Looks good. Just needs some comments.

**Comment by jasonadkison** [comment]
> Looks good. Just needs some comments.

@dijs I realized I couldn't easily explain what one-liner version was doing so I converted it into far simpler logic.

**Review by dijs — APPROVED** [review]

**Pull request merged**